### PR TITLE
Update README to mention pin_all_from for local modules

### PR DESCRIPTION
**Pull request by Rotario**
Whilst migrating from sprockets to propshaft, my relative imports stopped working due to the differences in asset fallbacks. Which I think Sprockets does and Propshaft doesn't.
The README doesn't yet mention pin_all_from which is really useful to know to get relative imports working from the import map.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,22 @@ It's in `app/javascript/application.js` you setup your application by importing
 
 It makes sense to use logical names that match the package names used by npm, such that if you later want to start transpiling or bundling your code, you won't have to change any module imports.
 
+### Local modules
+If you want to import local js module files from `app/javascript/src` or other sub-folders of `app/javascript` (such as `channels`), you must pin these to be able to import them.
+You can use `pin_all_from` to pick all files in a specific folder, so you don't have to `pin` each module individually.
+```rb
+#config/importmap.rb
```
Comment (radiantshaw) [inline README.md:71]:
```suggestion
# config/importmap.rb
```

File: README.md [hunk 2]
```diff
@@ -64,6 +64,22 @@ It's in `app/javascript/application.js` you setup your application by importing
...
+pin_all_from 'app/javascript/src`, under: 'src', to: 'src'
```
Comment (radiantshaw) [inline README.md:72]: If the `:under` option is a suffix of the first parameter, then the `:to` option is not needed. Maybe we can explain the difference between both the options as it was also not clear to me when I first started using the gem.

For more info, you can refer #67.

File: README.md [hunk 3]
```diff
@@ -64,6 +64,22 @@ It's in `app/javascript/application.js` you setup your application by importing
 
 It makes sense to use logical names that match the package names used by npm, such that if you later want to start transpiling or bundling your code, you won't have to change any module imports.
 
+### Local modules
+If you want to import local js module files from `app/javascript/src` or other sub-folders of `app/javascript` (such as `channels`), you must pin these to be able to import them.
+You can use `pin_all_from` to pick all files in a specific folder, so you don't have to `pin` each module individually.
```
Comment (radiantshaw) [inline README.md:69]:
```suggestion
You can use `pin_all_from` to pick all files in a specific folder, so you don't have to `pin` each module individually.

```

File: README.md [hunk 4]
```diff
@@ -64,6 +64,22 @@ It's in `app/javascript/application.js` you setup your application by importing
 
 It makes sense to use logical names that match the package names used by npm, such that if you later want to start transpiling or bundling your code, you won't have to change any module imports.
 
+### Local modules
```
Comment (radiantshaw) [inline README.md:67]:
```suggestion
### Local modules

```

File: README.md [hunk 5]
```diff
@@ -64,6 +64,22 @@ It's in `app/javascript/application.js` you setup your application by importing
...
+If you want to import local js module files from `app/javascript/src` or other sub-folders of `app/javascript` (such as `channels`), you must pin these to be able to import them.
+You can use `pin_all_from` to pick all files in a specific folder, so you don't have to `pin` each module individually.
+```rb
+#config/importmap.rb
+pin_all_from 'app/javascript/src`, under: 'src', to: 'src'
+```
+
+Allows you to
```
Comment (radiantshaw) [inline README.md:75]:
```suggestion
Allows you to:

```

File: README.md [hunk 6]
```diff
@@ -64,6 +64,22 @@ It's in `app/javascript/application.js` you setup your application by importing
...
+```js
+//app/javascript/application.js
```
Comment (radiantshaw) [inline README.md:77]:
```suggestion
// app/javascript/application.js
```

File: README.md [hunk 7]
```diff
@@ -64,6 +64,22 @@ It's in `app/javascript/application.js` you setup your application by importing
...
+import { ExampleFunction } from 'src/example_function'
+```
+Which imports the function from `app/javascript/src/example_function.js`.
+
+Note: Sprockets used to serve assets (albeit without filename digests) it couldn't find from the app/javascripts folder with logical relative paths, meaning pinning local files wasn't needed. Propshaft doesn't have this fallback, so when you use Propshaft you have to pin your local modules.
```
Comment (radiantshaw) [inline README.md:82]:
```suggestion
Note: Sprockets used to serve assets (albeit without filename digests) it couldn't find from the `app/javascripts` folder with logical relative paths, meaning pinning local files wasn't needed. Propshaft doesn't have this fallback, so when you use Propshaft you have to pin your local modules.
```

**Comment by morgoth** [comment]
@Rotario I find this info very useful. Are you able to finish the pull request?

File: README.md (hunk 2 above)
Comment (Rotario) [reply README.md:71]: Thanks, sorting now

**Comment by Rotario** [comment]
I've added the suggested changes

**Pull request merged**